core/logic_events.go

```diff
@@ -264,6 +264,14 @@ func (e eventsLogic) processWebToolsEvents() {
 		return
 	}
 
+	err = e.preProcessLocation(allWebToolsEvents)
+	if err != nil {
+		e.logger.Errorf("error on processing locations - %s", err)
+		return
+	}
+
+	location, _ := e.processLocation()
```

Comment:
Why don't you check the error?

Reply:
Also, this variables keeps many locations but not only one, so the name should match it.